packages/speech/src/index.js

```diff
@@ -0,0 +1,829 @@
+  if (!name) {
+    throw new Error('A name must be specified for an operation.');
+  }
+
+  return new common.GrpcOperation(this, name);
+};
+
+/**
+ * Perform synchronous speech recognition and receive results after all audio
+ * has been sent and processed. This is ideal for files 1 MB or below. For
+ * larger files, you will need to use {module:speech#startRecognition} or
+ * {module:speech#createRecognizeStream}.
+ *
+ * @resource [SyncRecognize API Reference]{@link https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.Speech.SyncRecognize}
+ * @resource [SyncRecognizeRequest API Reference]{@link https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.SyncRecognizeRequest}
+ *
+ * @param {string|object|module:storage/file} file - The source file to run the
+ *     detection on. It can be either a local file path, a remote file URL, a
+ *     Cloud Storage URI, a Cloud Storage File object, or a
+ *     [`RecognitionAudio`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionAudio)
+ *     object.
+ * @param {object} config - A `RecognitionConfig` object. See
+ *     [`RecognitionConfig`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig).
+ * @param {boolean=} config.verbose - Enable verbose mode for a more detailed
+ *     response. See the examples below. Default: `false`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {string|object[]} callback.results - By default, this will be a string
+ *     comprised of all of the transcriptions recognized from the audio. If
+ *     `config.verbose` is enabled, this is an object including a `transcript`
+ *     property, a `confidence` score from `0` - `100`, and an `alternatives`
+ *     array consisting of other transcription possibilities. See the examples
+ *     below for more.
+ * @param {object} callback.apiResponse - Raw API response. See
+ *     [`SyncRecognizeResponse`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#syncrecognizeresponse).
+ *
+ * @example
+ * var config = {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * };
+ *
+ * function callback(err, transcript, apiResponse) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // transcript = "how old is the Brooklyn Bridge"
+ * }
+ *
+ * //-
+ * // Run speech detection over a local file.
+ * //-
+ * speech.recognize('./bridge.raw', config, callback);
+ *
+ * //-
+ * // Run speech recognition over a file in Cloud Storage.
+ * //-
+ * speech.recognize('gs://your-bucket-name/bridge.raw', config, callback);
+ *
+ * //-
+ * // Run speech recognition over raw file contents.
+ * //-
+ * speech.recognize({
+ *   content: fs.readFileSync('./bridge.raw')
```

Comment:
Ah, my mistake I see that `object` is listed as an accepted type. Should we add a `file.content` param?